### PR TITLE
settings: introduce Service interface

### PR DESCRIPTION
**Pull request by keegancsmith**
This is a stub interface which is a proposal for how we use the settings API. The intention is to remove the currently exported functions. From reading call sites on the existing functions this should be easy to introduce.

Please look at the code which has comments and please give feedback. Based on that I will follow up with implementing it or closing.

Test Plan: CI

### CODE REVIEW

File: internal/settings/settings.go [hunk 1]
```diff
@@ -24,6 +24,34 @@ func defaultSettings() *schema.Settings {
...
+	//   c.settingsService.ForSubject(ctx, subj)
+	ForSubject(context.Context, api.SettingsSubject) (*schema.Settings, error)
+
+	// RelevantSubjects is the RelevantSubjects function but without the need
+	// to pass in a DB.
+	//
+	// Note: this only has one call site in graphql. I wonder if anyone
+	// actually reads this field anymore?
+	RelevantSubjects(context.Context, api.SettingsSubject) ([]api.SettingsSubject, error)
+}
```
Comment (stefanhengl) [inline internal/settings/settings.go:53]: I also see `Latest` in the code below. Do we need that as an additional method?

**Review by stefanhengl — COMMENTED** [review]
+1 for easy mocking and better readability

**Review by eseliger — COMMENTED** [review]
LGTM

**Review by camdencheek — APPROVED** [review]
Nice 😎

File: internal/settings/settings.go (hunk 1 above)
Comment (keegancsmith) [reply internal/settings/settings.go:53]: unexported here https://github.com/sourcegraph/sourcegraph/pull/53158

**Comment by keegancsmith** [comment]
alright I got started on migrating everything, but then realised its safer to just keep the old functions and introduce this service. So will land it, it is the PR from yesterday except with some cut paste + updated docstrings.

**Pull request merged**